core/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -107,6 +107,10 @@ protected void doExecute(Task task, SearchRequest searchRequest, ActionListener<
             searchRequest.preference());
         failIfOverShardCountLimit(clusterService, shardIterators.size());
 
+        if (searchRequest.source() != null) {
+            searchRequest.source().resolveIndexBoosts(indexNameExpressionResolver, clusterState, searchRequest.indicesOptions());
```

Comment:
can we move the resolution method out of `SearchSourceBuilder` too and also transport the info out of it as well? Like we do with aliasFilters map for instance? Does it make sense? Trying to keep `SearchSourceBuilder` as POJO as possible and have logic outside of it.